contracts/hydro/src/contract.rs

```diff
@@ -1745,6 +1766,280 @@ pub fn remove_token_info_provider(
         .add_attribute("provider_id", provider_id))
 }
 
+// Inserts or updates the drop info needed for conversion of lockups to dTokens
+// Validate that the sender is a whitelist admin
+// Sets the address of the drop core contract, the dToken denom, and the puppeteer address.
+fn set_drop_token_info(
+    deps: DepsMut<'_, NeutronQuery>,
+    _env: Env,
+    info: MessageInfo,
+    core_address: String,
+    d_token_denom: String,
+    puppeteer_address: String,
+) -> Result<Response<NeutronMsg>, ContractError> {
+    let whitelist_admins = WHITELIST_ADMINS.load(deps.storage)?;
+
+    if !whitelist_admins.contains(&info.sender) {
+        return Err(ContractError::Unauthorized);
+    }
+
+    let drop_token_info = DropTokenInfo {
+        address: deps.api.addr_validate(&core_address)?,
+        d_token_denom,
+        puppeteer_address: deps.api.addr_validate(&puppeteer_address)?,
+    };
+
+    DROP_TOKEN_INFO.save(deps.storage, &drop_token_info)?;
+
+    Ok(Response::new()
+        .add_attribute("action", "set_drop_token_info")
+        .add_attribute("sender", info.sender))
+}
+
+// Converts existing lockup (of lsm token) to dToken
+// The user specifies one or several lockups in order to convert it on drop
+// Hydro contract exectutes Bond message/s to drop core contract
+// https://github.com/hadronlabs-org/drop-contracts/blob/bdbb1a7986b4448aff10db3baaa150d71527e815/contracts/core/src/contract.rs#L1002-L1002
+// Drop core contract mints appropriate amount of dtoken given lsm shares sent
+pub fn convert_lockup_to_dtoken(
+    deps: DepsMut<NeutronQuery>,
+    _env: Env,
+    info: MessageInfo,
+    lock_ids: Vec<u64>,
+) -> Result<Response<NeutronMsg>, ContractError> {
+    let lock_ids_set: HashSet<u64> = lock_ids.into_iter().collect();
+
+    let lockups = query_user_lockups(
```

Comment:
maybe throwing an error if any of the provided lock_ids are not owned by the sender would be the better option?